mutiny-core/src/nostr/nwc.rs

```diff
@@ -166,99 +185,123 @@ impl NostrWalletConnect {
                             error: Some(NIP47Error {
-                                code: ErrorCode::InsufficantBalance,
-                                message: format!("Failed to pay invoice: {e}"),
+                                code: ErrorCode::QuotaExceeded,
+                                message: format!("Invoice amount too high: {msats} msats"),
                             }),
                             result: None,
-                        },
-                    }
-                } else {
-                    log_warn!(
-                        node_manager.logger,
-                        "Invoice amount too high: {msats} msats"
-                    );
-
-                    Response {
-                        result_type: Method::PayInvoice,
-                        error: Some(NIP47Error {
-                            code: ErrorCode::QuotaExceeded,
-                            message: format!("Invoice amount too high: {msats} msats"),
-                        }),
-                        result: None,
-                    }
-                };
-
-                let encrypted = encrypt(&server_key, &client_pubkey, content.as_json())?;
-
-                let p_tag = Tag::PubKey(event.pubkey, None);
-                let e_tag = Tag::Event(event.id, None, None);
-                let response =
-                    EventBuilder::new(Kind::WalletConnectResponse, encrypted, &[p_tag, e_tag])
-                        .to_event(&self.server_key)?;
-
-                return Ok(Some(response));
+                        }
+                    };
+
+                    let encrypted = encrypt(&server_key, &client_pubkey, content.as_json())?;
+
+                    let p_tag = Tag::PubKey(event.pubkey, None);
+                    let e_tag = Tag::Event(event.id, None, None);
+                    let response =
+                        EventBuilder::new(Kind::WalletConnectResponse, encrypted, &[p_tag, e_tag])
+                            .to_event(&self.server_key)?;
+
+                    return Ok((Some(response), needs_save));
+                }
+                SpendingConditions::RequireApproval => {
+                    let pending = PendingNwcInvoice {
+                        index: self.profile.index,
+                        invoice,
+                        event_id: event.id,
+                        pubkey: event.pubkey,
+                    };
+                    pending_nwc_lock.lock().await;
+
+                    let mut current: Vec<PendingNwcInvoice> = node_manager
+                        .storage
+                        .get_data(PENDING_NWC_EVENTS_KEY)?
+                        .unwrap_or_default();
+
+                    current.push(pending);
+
+                    node_manager
+                        .storage
+                        .set_data(PENDING_NWC_EVENTS_KEY, current, None)?;
+
+                    return Ok((None, needs_save));
+                }
```

Comment:
should this not be saved? or is the saving happening here instead of from the result of the call here?

Reply:
we dont need to update the nwc profile here, just save the invoice